File: csharpcore/Categories/AgedBrie.cs
namespace csharpcore.Categories
{
    public class AgedBrie : DefaultCategory
    {
        public override string Name => "Aged Brie";

        public override void UpdateQuality(Item item)
        {
            var factor = item.SellIn < 0 ? 2 : 1;
            item.Quality += factor;

            if (item.Quality > 50)
            {
                item.Quality = 50;
            }
        }
    }
}

File: Java/src/main/java/com/gildedrose/categories/AgedBrie.java
package com.gildedrose.categories;

import com.gildedrose.*;

public class AgedBrie extends DefaultCategory {

    @Override
    public String getName() {
        return "Aged Brie";
    }

    @Override
    public void UpdateQuality(Item item) {
        int factor = item.sellIn < 0 ? 2 : 1;
        item.quality += factor;

        if (item.quality > 50) {
            item.quality = 50;
        }
    }
}
